### PR TITLE
Make Pacbio model attributes non-nullable (#3858)(patch)

### PR DESCRIPTION
**Pull request by diitaz93**
## Description

Close https://github.com/Clinical-Genomics/add-new-tech/issues/124
Make all Pacbio database columns non-nullable and required

### Added

- Alembic migration script

### Changed

- Type of attributes of `PacbioSampleSequencingMetrics` and `PacbioSequencingRun` removing `None` from datatypes

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`

### How to test

- [x] Do the migration `alembic -c $ALEMBIC_CG_STAGE upgrade head`
- [x] Try to add an entry without the newly required attributes
- [x] Do the downgrade

### Expected test outcome

- [x] Check that the upgrade and downgrade work
- [x] You can't add a sample with null in the non-nullable attributes

## Review

- [x] Tests executed by SD
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Deployed to stage:
```shell

```
- [ ] Deployed to production:
```shell

```

### CODE REVIEW

**Comment by diitaz93** [comment]
## Tests
```shell
(cg-py3.11) (base) jsdiazboada@juasebmbp cg % alembic -c $ALEMBIC_CG_STAGE upgrade head
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade fd7c9d246255 -> 3503b53a9bc3, Make all Pacbio model attributes non-nullable
(cg-py3.11) (base) jsdiazboada@juasebmbp cg % 

```
Screenshot 2024-10-17 at 11 10 58
Screenshot 2024-10-17 at 11 11 30
Screenshot 2024-10-17 at 11 12 15
Screenshot 2024-10-17 at 11 12 45
Screenshot 2024-10-17 at 11 13 19
Screenshot 2024-10-17 at 11 13 41
Screenshot 2024-10-17 at 11 14 17
Screenshot 2024-10-17 at 11 14 42
Screenshot 2024-10-17 at 11 15 07
Screenshot 2024-10-17 at 11 15 37
Screenshot 2024-10-17 at 11 16 00

File: alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.alter_column(
+        table_name="pacbio_sequencing_run",
+        column_name="run_name",
+        existing_type=mysql.VARCHAR(length=64),
+        type_=sa.String(length=32),
```
Comment (islean) [inline alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py:26]: Is there a reason for changing the type here?

File: alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        table_name="pacbio_sequencing_run",
+        column_name="barcoded_hifi_reads",
+        existing_type=mysql.BIGINT(),
+        nullable=False,
+    )
+    op.alter_column(
+        table_name="pacbio_sequencing_run",
+        column_name="barcoded_hifi_reads_percentage",
+        existing_type=mysql.FLOAT(),
+        type_=sa.Numeric(precision=6, scale=2),
```
Comment (islean) [inline alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py:54]: Same question here

File: alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    op.alter_column(
+        table_name="pacbio_sequencing_run",
+        column_name="unbarcoded_hifi_mean_read_length",
+        existing_type=mysql.BIGINT(),
+        nullable=False,
+    )
+    op.alter_column(
+        table_name="pacbio_sample_run_metrics",
+        column_name="polymerase_mean_read_length",
+        existing_type=sa.BIGINT,
```
Comment (islean) [inline alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py:99]: What is the difference between sa.BIGINT and sa.BIGINT() here?

**Review by islean — APPROVED** [review]
Looks good! Just a bit confused about some of the type changes

File: alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py (hunk 1 above)
Comment (diitaz93) [reply alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py:26]: This was detected by alembic `autogenerate` and I left it as I found it relevant. In the model, this attribute is defined as `Mapped[Str32]` but it was originally defined in the database as `mysql.VARCHAR(length=64)`. Not a big deal but it is more correct now

File: alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py (hunk 2 above)
Comment (diitaz93) [reply alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py:54]: see above

File: alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py (hunk 3 above)
Comment (diitaz93) [reply alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py:99]: Typo, my bad. Should be sa.BIGINT(). Fixed

File: alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py (hunk 1 above)
Comment (diitaz93) [reply alembic/versions/2024_10_17_3503b53a9bc3_make_pacbio_attributes_non_nullable_.py:26]: resolved in person

**Pull request merged**